storage/buckets/buckets_test.go

```diff
@@ -272,6 +275,42 @@ func TestUniformBucketLevelAccess(t *testing.T) {
 	}
 }
 
+func TestLifecycleManagement(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	bucketName := tc.ProjectID + "-storage-buckets-tests"
+
+	ctx := context.Background()
+	testutil.CleanBucket(ctx, t, tc.ProjectID, bucketName)
+
+	if err := enableBucketLifecycleManagement(ioutil.Discard, bucketName); err != nil {
+		t.Fatalf("enableBucketLifecycleManagement: %v", err)
+	}
+
+	// verify lifecycle is set
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("storage.NewClient: %v", err)
+	}
+	attrs, err := client.Bucket(bucketName).Attrs(ctx)
+	if err != nil {
+		t.Fatalf("Bucket(%q).Attrs: %v", bucketName, err)
+	}
+
+	want := storage.LifecycleRule{
+		Action:    storage.LifecycleAction{Type: "Delete"},
+		Condition: storage.LifecycleCondition{AgeInDays: 100},
+	}
+
+	r := attrs.Lifecycle.Rules
+	if len(r) == 0 {
```

Comment:
Would len > 1 also indicate unexpected behavior? In this case I would check that as well.